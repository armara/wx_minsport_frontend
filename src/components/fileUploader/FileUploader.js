import React, { useState } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import FileBase64 from 'components/fileBase64/FileBase64'
import Button from 'components/button/Button'
import Input from 'components/input/Input'

import 'components/fileUploader/FileUploader.scss'

const FileUploader = ({ className, id, name, input: { onChange }, accept }) => {
  const [file, setFile] = useState({ name: '', base64: '' })

  return (
    <div className={classnames([className, 'file-uploader'])}>
      <Input
        value={file.name}
        className="form-input"
        id={id}
        name={name}
        disabled
      />
      <FileBase64
        accept={accept}
        onDone={f => {
          onChange(f)
          setFile(f)
        }}
      />
      <Button
        className="grey-control-button"
        onClick={() =>
          document.querySelectorAll('input[type="file"]')[0].click()
        }
      >
        Выбрать
      </Button>
    </div>
  )
}

FileUploader.defaultProps = {
  className: '',
  name: '',
  accept: null,
}

FileUploader.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string.isRequired,
  name: PropTypes.string,
  input: PropTypes.shape({
    onChange: PropTypes.func,
  }).isRequired,
  accept: PropTypes.string,
}

export default FileUploader
