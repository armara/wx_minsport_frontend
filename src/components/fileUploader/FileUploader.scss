@import "styles/mixins";

.file-uploader {
  display: flex;

  & > div {
    flex-grow: 1;
  }

  input[type="file"] {
    display: none;
  }
}
