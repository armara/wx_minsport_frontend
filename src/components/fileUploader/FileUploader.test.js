import React from 'react'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import FileUploader from 'components/fileUploader/FileUploader'

configure({ adapter: new Adapter() })

describe('<FileUploader />', () => {
  it('renders correctly default props', () => {
    const tree = renderer
      .create(<FileUploader id="test" input={{ onChange: () => {} }} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
