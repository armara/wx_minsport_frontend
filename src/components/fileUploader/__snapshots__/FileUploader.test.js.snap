// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileUploader /> renders correctly default props 1`] = `
<div
  className="file-uploader"
>
  <input
    className="form-input base-input input"
    disabled={true}
    id="test"
    name=""
    onChange={[Function]}
    placeholder=""
    type="text"
    value=""
  />
  <input
    accept={null}
    multiple={false}
    onChange={[Function]}
    type="file"
  />
  <button
    className="grey-control-button button base-button"
    disabled={false}
    id={null}
    onClick={[Function]}
    title=""
    type="button"
  >
    
    Выбрать
  </button>
</div>
`;
